src/Tmds.Ssh/SshAgent.cs

```diff
@@ -0,0 +1,282 @@
+// This file is part of Tmds.Ssh which is released under MIT.
+// See file LICENSE for full license details.
+
+using System.Buffers;
+using System.Buffers.Binary;
+using System.Diagnostics;
+using System.Net;
+using System.Net.Sockets;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace Tmds.Ssh
+{
+    // https://datatracker.ietf.org/doc/html/draft-miller-ssh-agent - SSH Agent Protocol
+    sealed class SshAgent : IDisposable
+    {
+        private const MessageId SSH_AGENTC_REQUEST_IDENTITIES = (MessageId)11;
+        private const MessageId SSH_AGENTC_SIGN_REQUEST = (MessageId)13;
+        private const MessageId SSH_AGENT_FAILURE = (MessageId)5;
+        private const MessageId SSH_AGENT_IDENTITIES_ANSWER = (MessageId)12;
+        private const MessageId SSH_AGENT_SIGN_RESPONSE = (MessageId)14;
+        private const uint SSH_AGENT_RSA_SHA2_256 = 2;
+        private const uint SSH_AGENT_RSA_SHA2_512 = 4;
+
+        private static EndPoint? _defaultEndPoint;
```

Comment:
@jborean93 for adding the named pipe support, we can probably change this from using an `EndPoint` to use a `string`. On Windows we will then assume this string is a named pipe address and on non-Windows it is a unix socket path.